src/lib/queues.ts

```diff
@@ -97,6 +98,43 @@ const worker = new Worker<SummarizeJobData | DreamJobData>(
         console.log(
           `Ingested conversation ${conversationId} for user ${userId}.`,
         );
+        
+        // Queue deep research job if there are messages
+        if (messages.length > 0) {
+          // Simple throttling: add a low probability to reduce job frequency
+          // This helps prevent too many jobs for users with many short conversations
+          if (Math.random() < env.DEEP_RESEARCH_PROBABILITY || 0.5) {
+            await batchQueue.add("deep-research", {
```

Comment:
Let's give this a job id too, and check if there's an existing job running, that way we don't create duplicate jobs. That's a second way of throttling. I want this to work really wwell.